File: NumberSteps_2052/NumberSteps.py

class Point:

    def __init__(self, x, y):
        self.x = x;
        self.y = y;

    def make_instance_from_raw_string(rawString):
        x = int(rawString.split(" ")[0]);
        y = int(rawString.split(" ")[1]);
        return Point(x,y);

    def __str__(self):
        return "Point has [x=" + self.x + ",y="+ self.y + "]";



class NumberSteps:

    def get_number_on(self,point):
        if point.x == point.y and self.__is_number_even(point.x):
            return point.x + point.y;
        if point.x == point.y and not self.__is_number_even(point.x):
            return point.x + point.y - 1;
        if point.x == (point.y + 2) and self.__is_number_even(point.x):
            return point.x + point.y
        if point.x == (point.y + 2) and not self.__is_number_even(point.x):
            return point.x + point.y - 1
        return "No Number";

    def __is_number_even(self,number):
        return number%2==0;



## Main section
numberSteps = NumberSteps();
countRepeats = int(input());
repeat = 0;
while repeat < countRepeats:
    rawString = input();
    point = Point.make_instance_from_raw_string(rawString);
    print(numberSteps.get_number_on(point));
    repeat+=1;

File: NumberSteps_2052/NumberSteps.c
#include<iostream>
using namespace std;

int main()
{
	int countRepeats = 0;
	int repeat = 0;

	cin >> countRepeats;
	while(repeat++ < countRepeats)
	{
		int x = 0;
		int y = 0;
	
		cin >> x;
		cin >> y;

		if(x==y && x%2==0)
		{
			cout << x+y << "\n";
		}
		else if(x==y && x%2!=0)
		{
			cout << x+y-1 << "\n";
		}
		else if(x==y+2 && x%2==0)
		{
			cout << x+y << "\n";
		}
		else if(x==y+2 && x%2!=0)
		{
			cout << x+y-1 << "\n";
		}
		else 
		{
			cout << "No Number" << "\n";
		}

	}
	
	return 0;
}
